app/authorization.go

```diff
@@ -37,7 +37,7 @@ func (a *App) SessionHasPermissionToChannel(session model.Session, channelId str
 		return false
 	}
 
-	cmc := a.Srv.Store.Channel().GetAllChannelMembersForUser(session.UserId, true)
+	cmc := a.Srv.Store.Channel().GetAllChannelMembersForUser(session.UserId, true, true)
```

Comment:
I'm concerned this will have unexpected side effects on other API endpoints that use this method for permissions checks.

Reply:
We can add an extra parameter to explicitly ask for permissions using deleted channels too. That way we can be sure that any other request (for update or delete, are not affected).

Reply:
:+1:

Reply:
I checked all the calls, and we only call this function from 3 sites.
1. In the authorization.go code (which is the reason of the change, and is working properly).
2. In the websocket connection to know if I have to send certain information, we can use "false" here or "true", doesn't matter too much, but "false" is safe.
3. in a function used only for tests called `IsUserInChannelUseCache`. And is find to set it to "false"

Reply:
Sorry, wasn't clear in my original comment. I meant that this might change the behaviour of `SessionHasPermissionToChannel` which we use in hundreds of places. What do you think?

Reply:
Yes, the final approach is:
1. You have permissions based on your membership, no matter if the channel is archived or not.
2. You can't update an archive channel except for "undelete" an archive channel (managed here: #diff-b9939ff9f50ada8fa7134b38684e1b72R306).

Basically that means that you can have permissions but it doesn't grant you to change the restrictions of the model. And now, archived channel edition is restricted from the model perspective (An archive channel enters in immutable mode as soon as you archive it).